core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -1607,13 +1607,32 @@ private CompiledQuery createTable(final ExecutionModel model, SqlExecutionContex
             throw SqlException.$(name.position, "table already exists");
         }
 
+        CharSequence volumeAlias = createTableModel.getVolumeAlias();
+        if (volumeAlias != null) {
+            CharSequence volumePath = configuration.getVolumeDefinitions().resolveAlias(volumeAlias);
+            if (volumePath != null) {
+                if (!ff.isDirOrSoftLinkDir(path.of(volumePath).$())) {
+                    throw CairoException.critical(0).put("not a valid path for volume [alias=").put(volumeAlias).put(", path=").put(path).put(']');
+                }
+            } else {
+                throw CairoException.critical(0).put("volume alias is not allowed [alias=").put(volumeAlias).put(']');
```

Comment:
I don't think it's critical. Should be even `SqlException` instead of `CairoException`.

Reply:
addressed in 1740a93da7c652c3501265c090db4de59996d0da